src/zonefile/inplace.rs

```diff
@@ -342,12 +358,27 @@ impl<'a> EntryScanner<'a> {
             self.zonefile.last_owner = Some(owner.clone());
         }
 
-        let class = match class {
-            Some(class) => {
-                self.zonefile.last_class = class;
+        let class = match (class, self.zonefile.last_class) {
+            (Some(class), Some(last_class)) => {
+                if self.zonefile.require_valid && class != last_class {
+                    return Err(EntryError::different_class(
+                        last_class, class,
+                    ));
+                }
+                class
+            }
+
+            (Some(class), None) => {
+                self.zonefile.last_class = Some(class);
                 class
             }
-            None => self.zonefile.last_class,
+
+            (None, Some(last_class)) => last_class,
+
+            (None, None) => {
```

Comment:
I think this is not correct. There is no default class according to RFC 1035. Do other parsers accept leaving out the class entirely?

Reply:
`ldns-read-zone` will happily read a zone that lacks the class and print it out as having class `IN`.

Reply:
@Philip-NLnetLabs: I updated this PR to raise an error if no last class exists and added the ability to set the last class explicitly. I also added some simple tests.

Reply:
One issue with this is it makes the library harder to use for the most common case, the IN class, and often we argue make it easy to use in the most common case.